mlrun/db/httpdb.py

```diff
@@ -138,17 +139,27 @@ def _enrich_and_validate(self, url):
             endpoint += f":{parsed_url.port}"
         base_url = f"{parsed_url.scheme}://{endpoint}{parsed_url.path}"
 
+        self.base_url = base_url
         username = parsed_url.username or config.httpdb.user
         password = parsed_url.password or config.httpdb.password
-
-        username, password, token = mlrun.platforms.add_or_refresh_credentials(
-            parsed_url.hostname, username, password, config.httpdb.token
-        )
-
-        self.base_url = base_url
         self.user = username
         self.password = password
-        self.token = token
+        self.token_provider = None
+
+        if config.auth_with_client_id:
+            token_endpoint = mlrun.get_secret_or_env("MLRUN_AUTH_TOKEN_ENDPOINT")
+            client_id = mlrun.get_secret_or_env("MLRUN_AUTH_CLIENT_ID")
+            client_secret = mlrun.get_secret_or_env("MLRUN_AUTH_CLIENT_SECRET")
```

Comment:
I would move this to the constructor e.g.
```
self.token_endpoint = token_endpoint or mlrun.get_secret_or_env("MLRUN_AUTH_TOKEN_ENDPOINT")
```

Reply:
You mean within the provider? I deliberately didn't want to have the provider understand what env variables are used to provide its configuration - it should do the OAuth work, not MLRun configuration.